src/EditorFeatures/CSharpTest/CodeActions/UseNamedArguments/UseNamedArgumentsTests.cs

```diff
@@ -199,5 +199,106 @@ class C : System.Attribute { public C(int arg1) {} public int P { get; set; } }"
 @"[C(arg1: 1, P = 2)]
 class C : System.Attribute { public C(int arg1) {} public int P { get; set; } }");
         }
+
+        [WorkItem(18848, "https://github.com/dotnet/roslyn/issues/18848")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNamedArguments)]
+        public async Task TestAvailableOnFirstTokenOfArgument1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int arg1, int arg2) 
+        => M([||]1 + 2, 2);
+}",
+@"class C
+{
+    void M(int arg1, int arg2)
+        => M(arg1: 1 + 2, arg2: 2);
+}");
+        }
+
+        [WorkItem(18848, "https://github.com/dotnet/roslyn/issues/18848")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNamedArguments)]
+        public async Task TestAvailableOnFirstTokenOfArgument2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(int arg1, int arg2) 
+        => M(1[||] + 2, 2);
+}",
+@"class C
+{
+    void M(int arg1, int arg2)
+        => M(arg1: 1 + 2, arg2: 2);
+}");
+        }
+
+        [WorkItem(18848, "https://github.com/dotnet/roslyn/issues/18848")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNamedArguments)]
+        public async Task TestAvailableOnFirstTokenOfArgument3()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+class C
+{
+    void M(Action arg1, int arg2) 
+        => M([||]() => { }, 2);
+}",
+@"
+using System;
+
+class C
+{
+    void M(Action arg1, int arg2) 
+        => M(arg1: () => { }, arg2: 2);
+}");
+        }
+
+        [WorkItem(18848, "https://github.com/dotnet/roslyn/issues/18848")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNamedArguments)]
+        public async Task TestMissingNotOnFirstTokenOfArgument2()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(int arg1, int arg2) 
+        => M(1 [||]+ 2, 2);
```

Comment:
💭 It seems like this change could result in some users believing we removed (or broke) this feature.